### PR TITLE
Windows Shebang

### LINKED ISSUE #233: windows: shebang (#!) scripts don't work
On Windows, running shebang scripts from xonsh generally doesn't work.  My preliminary investigation shows that _is_runnable_name() may be too restrictive and preventing the shebang-checking code from ever executing.  On Linux this is no big deal since Linux intrinsically can run shebang lines.  On Windows, this needs to be fixed.

I think a good answer here is for _is_runnable_name() to do an effective 'which' command to see if anything on the PATH may be runnable.  

For increased compatibility with msysGit, cygwin, and other unix-compatibility schemes on Windows, I think '#!' handling on Windows should additionally allow unix-style pathnames.  A simple way to do this is to check for a '#!' path starting with '/' and just take the basename of the path, relying on PATH lookup to find the right thing to execute.  Many '#!' scripts use '/usr/bin/env' anyway to perform a similar lookup of the correct file on the PATH.

Does this seem like a reasonable direction to go?

### PR DESCRIPTION
**Pull request by wrywerytwreywery**
This is a hack to get shebang lines working on Windows (should fix #233).  It does change the handling in *nix, too, but I think not in a way that has any ill effects.

### CODE REVIEW

**Comment by scopatz** [comment]
Thanks @wrywerytwreywery!

File: xonsh/built_ins.py [hunk 1]
```diff
@@ -301,9 +324,10 @@ def run_subproc(cmds, captured=True):
         stdout = last_stdout if cmd is last_cmd else PIPE
         uninew = cmd is last_cmd
         alias = builtins.aliases.get(cmd[0], None)
-        if _is_runnable_name(cmd[0]):
+        n = _get_runnable_name(cmd[0])
+        if n is not None:
             try:
-                aliased_cmd = get_script_subproc_command(cmd[0], cmd[1:])
+                aliased_cmd = get_script_subproc_command(n, cmd[1:])
```
Comment (scopatz) [inline xonsh/built_ins.py]: Ahh this is probably messing up running aliases...

File: xonsh/built_ins.py [hunk 2]
```diff
@@ -301,9 +324,10 @@ def run_subproc(cmds, captured=True):
         stdout = last_stdout if cmd is last_cmd else PIPE
         uninew = cmd is last_cmd
         alias = builtins.aliases.get(cmd[0], None)
-        if _is_runnable_name(cmd[0]):
+        n = _get_runnable_name(cmd[0])
```
Comment (scopatz) [inline xonsh/built_ins.py]: Or rather this is, because it doesn't check if the alias exists first.

Comment (wrywerytwreywery) [inline xonsh/built_ins.py]: Yeah, this is the one.

**Pull request merged**